assets/samplesheet_spaceranger_ffpe_cytassist.csv

```diff
@@ -0,0 +1,3 @@
+sample,fastq_1,fastq_2,tissue_hires_image,slide,area,manual_alignment
+Visium_FFPE_Human_Cervical_Cancer,https://github.com/nf-core/test-datasets/raw/modules/data/genomics/homo_sapiens/10xgenomics/spaceranger/human-brain-cancer-11-mm-capture-area-ffpe-2-standard_v2_ffpe_cytassist/CytAssist_11mm_FFPE_Human_Glioblastoma_2_S1_L001_R1_001.fastq.gz,https://github.com/nf-core/test-datasets/raw/modules/data/genomics/homo_sapiens/10xgenomics/spaceranger/human-brain-cancer-11-mm-capture-area-ffpe-2-standard_v2_ffpe_cytassist/CytAssist_11mm_FFPE_Human_Glioblastoma_2_S1_L001_R2_001.fastq.gz,https://github.com/nf-core/test-datasets/raw/modules/data/genomics/homo_sapiens/10xgenomics/spaceranger/human-brain-cancer-11-mm-capture-area-ffpe-2-standard_v2_ffpe_cytassist/CytAssist_11mm_FFPE_Human_Glioblastoma_image.tif,V52Y10-317,B1,
```

Comment:
Hi @fasterius, 

one discussion point here is how to structure the samplesheets. 
Previously, the (local) `spaceranger count` module took a directory of fastqs as input.

The new module on nf-core modules takes an explicit list of fastq files rather than a directory (this has been modelled after the `cellranger count` module). 

We have two options to handle this with respect to the samplesheet: 
 * keep `fastq_dir` as input and have the pipeline retreive files from within that directory
 * explicitly list `fastq_1` and `fastq_2` for each sample as shown in this mock CSV file. If a sample has multiple technical replicates (as it is the case in this example), there would need to be multiple rows with the same sample id pointing to all the different pairs of fastq files. 
 
Both is ok for me. On the one hand I like listing all files explicitly, on the other hand it  makes the samplesheet way more verbose and more cumbersome to create.

Reply:
Yeah, we have had quite a bit of discussions back-and-forth regarding the samplesheet. Using `fastq_dir` is indeed nice since it makes it easier for the end-user to create the samplesheet, but if the `cellranger count` uses explicit paths we might go with that for consistency. Thoughts, @cavenel ? I do like being explicit in general.

Reply:
We went with explicit files for the scrnaseq pipeline. But there this design choice was influenced by the fact that we wanted to support aligners other than cellranger. 

Since this pipeline is (and will likely stay) spaceranger only, using a directory might make more sense, because this is by 10x's design the input for spaceranger. 

Definitely happy to hear additional input by @cavenel!

Reply:
Could go either way, in other words!

Reply:
I never really liked having single files listed in the csv file, instead of one folder. Both for SpaceRanger input and for st_read_data. As SpaceRanger takes folder as input and gives folder as output, it seems more natural to use folders direclty.
But I agree it can be both ways, it would just be good to be consistent.

Reply:
It'd definitely be better to have caching for the directory that doesn't change when a new file is added, and I think this should be prioritized over end-user samplesheet convenience (as re-running many samples would be _quite_ the inconvenience) . I'd say that we either use @grst's solution with specifying a folder and get the files in there into a channel if it works, but if that doesn't work we specify files directly in the samplesheet. I do agree with @cavenel that it'd be nice to stick with a directory since that's how Space Ranger does it, but I'd prioritize the caching above that.

Reply:
To make matters worse, spaceranger also supports *multiple input folders* in case a sample was sequenced across multiple flow cells. In that case there can be duplicate file names, which doesn't allow to stage them in a single directory. 
See https://github.com/nf-core/scrnaseq/issues/147. 

I'll try to summarize in a separate issue the options we have. This  hasn't yet been solved in #scrnaseq either and applies there as well.

Reply:
I opened an issue in #scrnaseq for further discussion: https://github.com/nf-core/scrnaseq/issues/241

I don't expect there will be a quick solution, so for this PR I think I will ignore the special cases and just assume that specifying one single folder will do the trick (likely covers >95% of the usecases anyway).